### PR TITLE
delete images and add coverage gem

### PR DESCRIPTION
**Pull request by kaozdl**
#### Trello ticket:

------
#### How I solved it:
manejo los updates por separado para tener en cuenta el caso en el que me manden _destroy

------
#### Refactors or changes not directly related to the main purpose of this PR:

------
#### Screenshots:

------

### CODE REVIEW

File: spec/requests/api/v1/admin/communications/update_spec.rb [hunk 1]
```diff
@@ -51,6 +51,20 @@
           expect(bd_communication.image.attached?).to eq true
         end
 
+        it 'deattaches a file to a communication on update' do
+          data = { 'communication': { 'title': 'Lala', 'image': { 'data': open_file_encoded('photo.jpg') } } }
+          put "/api/v1/admin/communications/#{communication.id}", headers: auth_headers, params: data
+          expect(response).to have_http_status 200
+          bd_communication = Communication.first
+          expect(bd_communication.title).to eq 'Lala'
+          expect(bd_communication.image.attached?).to eq true
```
[the remark this answers is not in the document]
Comment (kaozdl) [reply spec/requests/api/v1/admin/communications/update_spec.rb:60]: el tema es que hay que crearla con imagen. y no tengo muy claro como se hace.
Antes de mandar eso sin un test preferí esto

**Review by CMaidana — DISMISSED** [review]

File: app/controllers/api/v1/admin/communications_controller.rb [hunk 2]
```diff
@@ -34,6 +35,19 @@ def destroy
 
         private
 
+        def update_params
+          params.require(:communication).permit(:title, :text, :published, :recurrent_on)
```
Comment (DJA89) [inline app/controllers/api/v1/admin/communications_controller.rb:39]: Just to be sure, you can't make a communication recurrent once created?

File: app/controllers/api/v1/admin/communications_controller.rb [hunk 3]
```diff
@@ -34,6 +35,19 @@ def destroy
...
+        end
+
+        def handle_attachments
+          return unless params[:communication][:image]
+
+          @communication.image.purge
```
Comment (DJA89) [inline app/controllers/api/v1/admin/communications_controller.rb:45]: We just need to do this if we want to destroy it, right?

**Comment by DJA89** [comment]
Se agrego a lo que se corre en github el simplecov?

File: app/controllers/api/v1/admin/communications_controller.rb (hunk 3 above)
Comment (kaozdl) [reply app/controllers/api/v1/admin/communications_controller.rb:45]: no, we need to do it always.
if we don't even if the files are detacched, they remain in S3 wich could be a problem

**Review by guzmanmo — DISMISSED** [review]

File: app/controllers/api/v1/admin/communications_controller.rb (hunk 2 above)
Comment (kaozdl) [reply app/controllers/api/v1/admin/communications_controller.rb:39]: my bad

**Comment by kaozdl** [comment]
No se agregó no.

File: Gemfile [hunk 4]
```diff
@@ -64,6 +64,7 @@ end
 group :test do
   gem 'database_cleaner', '~> 1.7'
   gem 'shoulda-matchers', '~> 3.1'
+  gem 'simplecov', require: false, group: :test
```
Comment (DJA89) [inline Gemfile:67]: This is already on the test group, it doesn't need it specified

**Review by DJA89 — CHANGES_REQUESTED** [review]

**Review by facundotorterola — APPROVED** [review]

**Review by guzmanmo — APPROVED** [review]

**Pull request merged**